### DISCUSSION TITLE
getopt.h

### DISCUSSION THREAD

**Discussion by Ank1taS** [Blaze-Stars/Image-Filter-Application#20] [Q&A] [answered]
Use of `getopt.h` header. refer to [this](https://www.mankier.com/3/getopt)
`getopt()` is used to parse command-line arguments.
please discuss more for help

**Comment by chandrakant100** [comment, marked as answer]
The ```getopt()``` function is a builtin function in C and is used to parse command line arguments.
```
getopt(int argc, char *const argv[], const char *optstring)

`optstring` is simply  a list of characters, 
each representing a single character option.
```
__Return Value:__ The getopt() function returns different values:

+ If the option takes a value, that value is `pointer`to the external variable `optarg`.
+ `-1` if there are no more options to process.
+ `?`when there is an unrecognized option and it stores into external variable optopt.
+ If an option requires a value (such as -f in our example) and no value is given, getopt normally returns `?`.
    By placing a colon as the first character of the options string, getopt returns: instead of `?` when no value is given.

src -> [link](https://www.geeksforgeeks.org/getopt-function-in-c-to-parse-command-line-arguments).
credits: [geeksforgeeks](https://www.geeksforgeeks.org)

**Comment by Ank1taS** [comment]
#### RETURN VALUE

- The getopt() function shall return the next option character specified on the command line.
-  A colon ( ':' ) shall be returned if getopt() detects a missing argument and the first character of optstring was a colon ( ':' ).
-  A question mark ( '?' ) shall be returned if getopt() encounters an option character, not in optstring or detects a missing argument and the first character of optstring was not a colon ( ':' ).
-  Otherwise, getopt() shall return -1 when all command-line options are parsed.


https://pubs.opengroup.org/onlinepubs/009696799/functions/getopt.html
